commit 5334e7bc0b626cbbbd155e559d931cd6b2e49630 (refs/changes/09/1109/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-12-12T11:15:47-08:00 (2 years, 10 months ago)
    
    TOOLS-1633 'sdcadm up' of assets broken by TOOLS-1591
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 151acd8..fc1a9fd 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,11 @@
 
 # sdcadm Changelog
 
+## 1.15.1
+
+- TOOLS-1633: Fix 'sdcadm up' of the assets zone. It was broken in
+  version 1.15.0.
+
 ## 1.15.0
 
 - TOOLS-1591: 'sdcadm up' of a number of core Triton services (the simple
diff --git a/lib/procedures/update-stateless-services-v1.js b/lib/procedures/update-stateless-services-v1.js
index 4056a50..2720e56 100644
--- a/lib/procedures/update-stateless-services-v1.js
+++ b/lib/procedures/update-stateless-services-v1.js
@@ -101,19 +101,24 @@ UpdateStatelessServicesV1.prototype.execute = function ussv1Execute(opts, cb) {
             /*
              * If the service params require a delegate dataset, then ensure
              * the instance has one.
+             *
+             * Also another workaround for assets, which has a limited hacked
+             * in 'service' object.
              */
-            assert.optionalBool(change.service.params.delegate_dataset,
-                'change.service.params.delegate_dataset');
-            if (change.service.params.delegate_dataset) {
-                steps.push(function (_, next) {
-                    s.ensureDelegateDataset({
-                        service: change.service,
-                        progress: opts.progress,
-                        zonename: change.inst.zonename,
-                        log: opts.log,
-                        server: change.inst.server
-                    }, next);
-                });
+            if (change.service.params) {
+                assert.optionalBool(change.service.params.delegate_dataset,
+                    'change.service.params.delegate_dataset');
+                if (change.service.params.delegate_dataset) {
+                    steps.push(function (_, next) {
+                        s.ensureDelegateDataset({
+                            service: change.service,
+                            progress: opts.progress,
+                            zonename: change.inst.zonename,
+                            log: opts.log,
+                            server: change.inst.server
+                        }, next);
+                    });
+                }
             }
 
             steps = steps.concat([
diff --git a/package.json b/package.json
index 38164a2..fe744b9 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.15.0",
+  "version": "1.15.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
